tests/torch/test_loggers.py

```diff
@@ -0,0 +1,111 @@
+# --------------------------------------------------------------------------------------
+# Part of the interTwin Project: https://www.intertwin.eu/
+#
+# Created by: Anna Elisa Lappe
+#
+# Credit:
+# - Anna Lappe <anna.elisa.lappe@cern.ch> - CERN
+# -------------------------------------------------------------------------------------
+
+from argparse import Namespace
+from unittest.mock import MagicMock
+
+import pytest
+from lightning.pytorch.callbacks.model_checkpoint import ModelCheckpoint
+
+from itwinai.loggers import Logger
+from itwinai.torch.loggers import PyTorchLightningLogger
+
+
+@pytest.fixture
+def lightning_loggers():
+    """Setup a generic PyTorchLightningLogger with mock loggers for testing."""
+    itwinai_logger_mock = MagicMock()
+    lightning_logger = PyTorchLightningLogger(itwinai_logger=itwinai_logger_mock)
+
+    return itwinai_logger_mock, lightning_logger
+
+
+def test_lightning_experiment_property(lightning_loggers):
+    """Test that the experiment property initializes
+    the logger context and returns the logger."""
+    itwinai_logger_mock, lightning_logger = lightning_loggers
+    experiment = lightning_logger.experiment
+    assert isinstance(experiment, Logger)
+    itwinai_logger_mock.create_logger_context.assert_called_once_with(rank=0)
+
+
+def test_lightning_log_metrics(lightning_loggers):
+    """Test log_metrics method to ensure it logs metrics with the correct parameters."""
+    itwinai_logger_mock, lightning_logger = lightning_loggers
+    metrics = {"accuracy": 0.95, "loss": 0.1}
+    step = 1
+    lightning_logger.log_metrics(metrics=metrics, step=step)
+    for key, value in metrics.items():
+        itwinai_logger_mock.log.assert_any_call(
+            item=value, identifier=key, kind='metric', step=step)
+
+
+def test_lightning_log_hyperparams(lightning_loggers):
+    """Test log_hyperparams method to ensure it logs hyperparameters correctly."""
+    itwinai_logger_mock, lightning_logger = lightning_loggers
+    dict_params = {"learning_rate": 0.001, "batch_size": 32}
+    namespace_params = Namespace(**dict_params)
```

Comment:
to avoid confusion I would move this line below before `namespace_params` is actually used